File: apps/api/src/domain/repositories/task_repository.py
"""
Domain Repository Interface: Task Repository
Defines the contract for task data access
"""

from abc import ABC, abstractmethod
from typing import List, Optional
from uuid import UUID

from src.domain.entities.task import Task, TaskStatus


class TaskRepository(ABC):
    """Abstract task repository"""

    @abstractmethod
    async def find_by_id(self, task_id: UUID) -> Optional[Task]:
        """Find task by ID"""
        pass

    @abstractmethod
    async def find_by_status(self, status: TaskStatus) -> List[Task]:
        """Find tasks by status"""
        pass

    @abstractmethod
    async def find_by_assignee(self, assignee_id: UUID) -> List[Task]:
        """Find tasks by assignee"""
        pass

    @abstractmethod
    async def find_all(self) -> List[Task]:
        """Find all tasks"""
        pass

    @abstractmethod
    async def save(self, task: Task) -> None:
        """Save task"""
        pass

    @abstractmethod
    async def delete(self, task_id: UUID) -> None:
        """Delete task"""
        pass

File: apps/web/src/domain/repositories/TaskRepository.ts
// Domain Repository Interface: Task Repository
// Defines the contract for task data access

import { Task, TaskId, TaskStatus } from '../entities/Task'

export interface TaskRepository {
  findById(id: TaskId): Promise<Task | null>
  findByStatus(status: TaskStatus): Promise<Task[]>
  findByAssignee(assigneeId: string): Promise<Task[]>
  findAll(): Promise<Task[]>
  save(task: Task): Promise<void>
  delete(id: TaskId): Promise<void>
}
